include/openenclave/enclave.h

```diff
@@ -504,6 +504,18 @@ oe_enclave_t* oe_get_enclave(void);
  */
 oe_result_t oe_random(void* data, size_t size);
 
+/**
+ * Enable enable OE SDK logging for enclave code path
+ *
+ * This function lets an enclave to opt-in enclave SDK logging
+ *
+ * @param enabled: when set to true, this parameter tells OE SDK to enable
+ * logging enclave related operations set to false to disbale it. The default
+ * behavior is disabled.
+ *
+ */
+void oe_enable_sdk_logging(bool enabled);
```

Comment:
I don't understand why this would be simply boolean, as opposed to the log level, log provider, etc.
I also don't understand why we would make enabling it public but the ability to write log entries only be internal.

Reply:
An enclave can use this API to opt-in/out SDK logging for the enclave operations
void oe_enable_sdk_logging(bool enabled) for opt-in and opt-out enclave logging
Note: This API supersedes the effect of OE_ENCLAVE_FLAG_DEBUG flag on an enclave. It enables enclave developers to better control on what to share and what not to share in case that collaboration debugging scenarios arise.

For example, on a debug-enclave, you can still block logging for certain enclave code sections if an developer only wants to share limited logging information with collaboration partners.

oe_enable_sdk_logging(false)

// no enclave logging from this block

oe_enable_sdk_logging(true)

Reply:
I'm not convinced that this is a good mechanism for exerting this level of control. 
* Consider the reverse case: it also allows an unwary developer to force enable logging of OE SDK in code which cannot be overridden by the runtime environment owner (e.g. Azure admin).
* If the expectation is that the API is intended specifically for block scoping, it's not currently implemented to behave that way; you could enable/disable it and forget to disable/enable it which now affects global functionality. 
* It assumes that a dev understands a static scoping of SDK logging information at build time rather than the operator choosing to collect a logging subset of information at runtime. From a scenario perspective, it's likely that any scoping done by an app developer over SDK internal logging is going to create a bunch of confusion for the consumers of the resulting log (i.e. the OE maintainers who consume those logs.) since the service operator and us would not have visibility into what was enabled/disabled.

I would recommend removing this interface entirely. From an app developer perspective, I would assert that they only need the ability to globally disable OE SDK logging.

Reply:
I went thought a couple of usage  scenarios with CodeMonkeyLeet and I agreed to remove oe_enable_sdk_logging from the the proposal.

Thanks for the great feedback,CodeMonkeyLeet.